AutoGear.lua

```diff
@@ -1889,80 +1891,99 @@ AutoGearFrame:SetScript("OnEvent", function (this, event, arg1, arg2, arg3, arg4
-				for i = 1, quests do
-					local title, questLevel, isTrivial, frequency, repeatable, isComplete, isLegendary = info[(i-1)*7+1], info[(i-1)*7+2], info[(i-1)*7+3], info[(i-1)*7+4], info[(i-1)*7+5], info[(i-1)*7+6], info[(i-1)*7+7]
-					if (not isTrivial) then
-						C_GossipInfo.SelectAvailableQuest(i)
+			end			
+			--available quests			
+			if (SL) then 
+				for i = 1, C_GossipInfo.GetNumAvailableQuests() do
+					local quest = C_GossipInfo.GetAvailableQuests()[i]
+					if (quest["isTrivial"]==false) then
+	  					C_GossipInfo.SelectAvailableQuest(i)
 					end
 				end
-			else
-				quests = GetNumGossipAvailableQuests()
+			end
+			if (BFA or IsClassic) then
 				info = {GetGossipAvailableQuests()}
-				for i = 1, quests do
-				local name, level, isTrivial, frequency, isRepeatable = info[(i-1)*7+1], info[(i-1)*7+2], info[(i-1)*7+3], info[(i-1)*7+4], info[(i-1)*7+5]
+				for i = 1, GetNumGossipAvailableQuests() do
+					local name, level, isTrivial, frequency, isRepeatable = info[(i-1)*7+1], info[(i-1)*7+2], info[(i-1)*7+3], info[(i-1)*7+4], info[(i-1)*7+5]
 					if (not isTrivial) then
 						SelectGossipAvailableQuest(i)
 					end
 				end
 			end
 		elseif (event == "QUEST_GREETING") then
 			--active quests
-			if (not IsClassic) then
-				local quests = C_GossipInfo.GetNumActiveQuests()
-				for i = 1, quests do
+			if (SL) then
+				for i = 1, C_GossipInfo.GetNumActiveQuests() do
+					local quest = C_GossipInfo.GetActiveQuests()[i]
+					if (quest["isComplete"]==true) then
+						C_GossipInfo.SelectActiveQuest(i)
+					end
+				end
+			end
+			if (SL and GetNumActiveQuests() > 0) then 
+				for i = 1, GetNumActiveQuests() do
 					local title, isComplete = GetActiveTitle(i)
 					if (isComplete) then
-						C_GossipInfo.SelectActiveQuest(i)
+						SelectActiveQuest(i)
 					end
 				end
-			else
-				local quests = GetNumActiveQuests()
-				for i = 1, quests do
+			end
+			if (IsClassic or BFA) then
+				for i = 1, GetNumActiveQuests() do
 					local title, isComplete = GetActiveTitle(i)
 					if (isComplete) then
 						SelectActiveQuest(i)
 					end
 				end
 			end
-			--available quests
-			if (not IsClassic) then 
-				quests = C_GossipInfo.GetNumAvailableQuests()
-				for i = 1, quests do
-					local isTrivial, isDaily, isRepeatable = GetAvailableQuestInfo(i)
+			--available quests			
+			if (SL) then 
+				if (C_GossipInfo.GetNumAvailableQuests() > 0) then
+					for i = 1, C_GossipInfo.GetNumAvailableQuests() do
+						local quest = C_GossipInfo.GetAvailableQuests()[i]
+						if (quest["isTrivial"]==false) then
+	  						C_GossipInfo.SelectAvailableQuest(i)
+						end
+					end
+				elseif (C_GossipInfo.GetNumAvailableQuests() == 0) then
+					for i = 1, GetNumAvailableQuests() do
+                				local isTrivial, frequency, isRepeatable, isLegendary, questID = GetAvailableQuestInfo(i)
+						if (not isTrivial) then
+							SelectAvailableQuest(i)
+						end
+					end
+				end
+			end
+			if (BFA) then 
```

Comment:
Please rename to ```IsBFA``` to follow the convention and use ```elseif``` instead of ```end``` and ```if``` to clarify exclusivity.